interfaces/inc/c_pal/srw_lock_ll.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+
+#ifndef SRW_LOCK_H
+#define SRW_LOCK_H
+
+#ifdef __cplusplus
+#include <cstdbool>
+#else
+#include <stdbool.h>
+#endif
+
+#include "macro_utils/macro_utils.h"
+#include "umock_c/umock_c_prod.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct SRW_LOCK_HANDLE_DATA_TAG* SRW_LOCK_HANDLE;
+
+#define SRW_LOCK_TRY_ACQUIRE_RESULT_VALUES \
+    SRW_LOCK_TRY_ACQUIRE_OK, \
+    SRW_LOCK_TRY_ACQUIRE_COULD_NOT_ACQUIRE, \
+    SRW_LOCK_TRY_ACQUIRE_INVALID_ARGS
+
+MU_DEFINE_ENUM(SRW_LOCK_TRY_ACQUIRE_RESULT, SRW_LOCK_TRY_ACQUIRE_RESULT_VALUES)
+
+MOCKABLE_FUNCTION(, SRW_LOCK_HANDLE, srw_lock_create, bool, do_statistics, const char*, lock_name);
```

Comment:
>bool, do_statistics, [](http://example.com/codeflow?start=54&length=20)

this parameter is not part of `srw_lock_ll_requirements.md`

Reply:
I am confused why this file exists in this PR.

Reply:
I think this is part of the interface... cause I see it mentioned in cmakelists.txt of "interfaces" folder.

Still puzzled./

Reply:
It was supposed to have the APIs in the .md, but someone f-ed up and did not put them in here. Fixed.